### PR TITLE
eth/catalyst: use setcanonical instead of sethead in simulated fork

### PR DESCRIPTION
**Pull request by holiman**
Fixes https://github.com/ethereum/go-ethereum/issues/30448

### CODE REVIEW

**Comment by MariusVanDerWijden** [comment]
Hmm looks like a test for the simulated backend fails with this

**Comment by holiman** [comment]
In order to fix the failing test, I had to change the test semantics. In the old behaviour, it expected a dropped block to simply disappear, along with all txs. Now, however, we expect it to use the fork-events, where the tx pool actually picks up transactions that were dropped in a dropped head. 
Thus, when we mine a new block on top, the tx will be included without us needing to do anything. So the test changes from one behaviour to another, and thus becomes a totally different test, really .

I'm not 100% sure if the changed test is ok, or if we should replace/fix it some other way. cc @karalabe ?

**Review by rjl493456442 — APPROVED** [review]
sgtm

File: ethclient/simulated/backend_test.go [hunk 1]
```diff
@@ -228,15 +228,13 @@ func TestForkResendTx(t *testing.T) {
 	if err := sim.Fork(parent.Hash()); err != nil {
 		t.Errorf("forking: %v", err)
 	}
-
 	// 5.
 	sim.Commit()
 	if err := client.SendTransaction(ctx, tx); err != nil {
```
Comment (zhiqiangxu) [inline ethclient/simulated/backend_test.go:233]: It seems `SendTransaction` can be omitted if the `tx` is expected to be picked up automatically?

Comment (holiman) [reply ethclient/simulated/backend_test.go:233]: Good point, thanks

**Review by MariusVanDerWijden — APPROVED** [review]
LGTM

**Pull request merged**